contracts/token/SRC20.sol

```diff
@@ -578,4 +501,58 @@ contract SRC20 is ISRC20, ISRC20Owned, ISRC20Managed, SRC20Detailed, Featured,
 
         emit Approval(owner, spender, value);
     }
+
+    /**
+     * Perform multiple token transfers from the token owner's address.
+     * The tokens should already be minted. If this function is to be called by
+     * an actor other than the owner (a delegate), the owner has to call approve()
+     * first to set up the delegate's allowance.
+     *
+     * @param _addresses an array of addresses to transfer to
+     * @param _values an array of values
+     * @return True on success
+     */ 
+   function bulkTransfer (
+        address[] calldata _addresses, uint256[] calldata _values) external onlyDelegate returns (bool) {
+        require(_addresses.length == _values.length, "Input dataset length mismatch");
+
+        uint256 count = _addresses.length;
+        for (uint256 i = 0; i < count; i++) {
+            address to = _addresses[i];
+            uint256 value = _values[i];
+            _approve(owner(), msg.sender, _allowances[owner()][msg.sender].sub(value));
+            _transfer(owner(), to, value);
```

Comment:
Here, `require(!_features.checkTransfer(from, to));` will be in effect, not sure if that's intention - ie. bulkTransfer to frozen addresses will be reverted. Same in encoded version.

Reply:
Agree.